# Chicago City Data Analysis: Socioeconomic Indicators & Crime Trends

**Author:** Zaara Shaikh<br>
**Project Type:** Database Analysis & Case Study<br>
**Tech Stack:** SQL (SQLite), Python, Pandas

## Executive Summary
This project analyzes reports from the Chicago Police Department, Chicago Public Schools, and City Census Data to identify relationships between socioeconomic hardship and criminal activity. By utilizing SQL for data extraction and aggregation, this analysis aims to answer critical questions regarding community safety, school performance, and public health indicators.

## Business Objectives
The analysis focuses on three core areas:
1.  **Crime Volume:** Identifying high-frequency crime types and their geographic concentration.
2.  **Socioeconomic Correlation:** Investigating how per capita income and hardship indices correlate with crime rates.
3.  **Education & Safety:** Assessing the safety scores of public schools and the prevalence of crimes involving minors.

### Datasets Overview

In this assignment, we used three datasets that are available on the city of Chicago's Data Portal:
1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>


### Download the datasets

1. <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Census Data</a>

2. <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>

3. <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### Installing Libraries

In [1]:
!pip install pandas
!pip install ipython-sql prettytable 

import prettytable

prettytable.DEFAULT = 'DEFAULT'

Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


### Store the datasets in database tables
To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables as under:
1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

In [2]:
#Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`
import sqlite3
import pandas as pd
%load_ext sql

In [3]:
#Load the SQL magic module
%sql sqlite:///FinalDB.db

In [4]:
# Load Socioeconomic Indicators dataset
socio_df = pd.read_csv("https://data.cityofchicago.org/resource/kn9c-c2s2.csv")
socio_df.to_sql("CENSUS_DATA", con=sqlite3.connect("FinalDB.db"), if_exists="replace", index=False)

# Load Chicago Public Schools dataset
school_df = pd.read_csv("https://data.cityofchicago.org/resource/9xs2-f89t.csv")
school_df.to_sql("CHICAGO_PUBLIC_SCHOOLS", con=sqlite3.connect("FinalDB.db"), if_exists="replace", index=False)

# Load Chicago Crimes dataset
crime_df = pd.read_csv("https://data.cityofchicago.org/resource/ijzp-q8t2.csv")
crime_df.to_sql("CHICAGO_CRIME_DATA", con=sqlite3.connect("FinalDB.db"), if_exists="replace", index=False)


1000

In [5]:
#Establish a connection between SQL magic module and the database `FinalDB.db`
%sql sqlite:///FinalDB.db

## Problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [6]:
%%sql SELECT COUNT(*) AS TOTAL_CRIMES
FROM CHICAGO_CRIME_DATA;

 * sqlite:///FinalDB.db
Done.


TOTAL_CRIMES
1000


### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [7]:
import sqlite3
con = sqlite3.connect("FinalDB.db")
pd.read_sql_query("PRAGMA table_info(CENSUS_DATA);", con)

,cid,name,type,notnull,dflt_value,pk
0,0,ca,REAL,0,None,0
1,1,community_area_name,TEXT,0,None,0
2,2,percent_of_housing_crowded,REAL,0,None,0
3,3,percent_households_below_poverty,REAL,0,None,0
4,4,percent_aged_16_unemployed,REAL,0,None,0
5,5,percent_aged_25_without_high_school_diploma,REAL,0,None,0
6,6,percent_aged_under_18_or_over_64,REAL,0,None,0
7,7,per_capita_income_,INTEGER,0,None,0
8,8,hardship_index,REAL,0,None,0


In [8]:
%%sql
    SELECT community_area_name, per_capita_income_
    FROM CENSUS_DATA
    WHERE per_capita_income_ < 11000;

 * sqlite:///FinalDB.db
Done.


community_area_name,per_capita_income_
West Garfield Park,10934
South Lawndale,10402
Fuller Park,10432
Riverdale,8201


### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [9]:
%%sql 
    SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION
    FROM CHICAGO_CRIME_DATA
    WHERE DESCRIPTION LIKE '%MINOR%'
    AND DESCRIPTION NOT LIKE '%CHILD%';

 * sqlite:///FinalDB.db
Done.


case_number,primary_type,description
JJ503831,BATTERY,"AGGRAVATED P.O. - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ503747,ASSAULT,"PROTECTED EMPLOYEE - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ503696,BATTERY,"AGGRAVATED P.O. - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ503634,BATTERY,"AGGRAVATED P.O. - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ504293,BATTERY,"AGGRAVATED P.O. - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ503478,LIQUOR LAW VIOLATION,SELL / GIVE / DELIVER LIQUOR TO MINOR
JJ503259,BATTERY,"AGGRAVATED - HANDS, FISTS, FEET, NO / MINOR INJURY"
JJ502687,BATTERY,"AGGRAVATED P.O. - HANDS, FISTS, FEET, NO / MINOR INJURY"


### Problem 4

##### List all kidnapping crimes involving a child?


In [23]:
%%sql 
    SELECT *
    FROM CHICAGO_CRIME_DATA
    WHERE PRIMARY_TYPE = 'KIDNAPPING' AND DESCRIPTION LIKE '%CHILD%'; -- No data on kidnapping involving a child because there is only one kidnapping case comment out the "AND DESCRIPTION..." part and check

 * sqlite:///FinalDB.db
Done.


id,case_number,date,block,iucr,primary_type,description,location_description,arrest,domestic,beat,district,ward,community_area,fbi_code,x_coordinate,y_coordinate,year,updated_on,latitude,longitude,location


### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [11]:
%%sql
    SELECT DISTINCT PRIMARY_TYPE
    FROM CHICAGO_CRIME_DATA
    WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%';

 * sqlite:///FinalDB.db
Done.


primary_type
CRIMINAL TRESPASS
MOTOR VEHICLE THEFT


### Problem 6

##### List the type of schools along with the average safety score for each type.


In [12]:
%%sql
    SELECT "Elementary, Middle, or High School", AVG(SAFETY_SCORE) AS AVERAGE_SAFETY_SCORE
    FROM CHICAGO_PUBLIC_SCHOOLS
    GROUP BY "Elementary, Middle, or High School";

 * sqlite:///FinalDB.db
Done.


"""Elementary, Middle, or High School""",AVERAGE_SAFETY_SCORE
"Elementary, Middle, or High School",49.50487329434698


### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [13]:
%%sql
    SELECT COMMUNITY_AREA_NAME, PERCENT_HOUSEHOLDS_BELOW_POVERTY
    FROM CENSUS_DATA
    ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC
    LIMIT 5;

 * sqlite:///FinalDB.db
Done.


community_area_name,percent_households_below_poverty
Riverdale,56.5
Fuller Park,51.2
Englewood,46.6
North Lawndale,43.1
East Garfield Park,42.4


### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [14]:
%%sql
    SELECT "COMMUNITY_AREA_NUMBER", COUNT(*) as crime_count
    FROM CHICAGO_CRIME_DATA
    GROUP BY "COMMUNITY_AREA_NUMBER"
    ORDER BY crime_count DESC
    LIMIT 1;

 * sqlite:///FinalDB.db
Done.


"""COMMUNITY_AREA_NUMBER""",crime_count
COMMUNITY_AREA_NUMBER,1000


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [15]:
%%sql 
    SELECT COMMUNITY_AREA_NAME
    FROM CENSUS_DATA
    WHERE HARDSHIP_INDEX = (SELECT MAX(HARDSHIP_INDEX) FROM CENSUS_DATA);

 * sqlite:///FinalDB.db
Done.


community_area_name
Riverdale


### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [16]:
%%sql
    SELECT community_area_name
    FROM CENSUS_DATA
    WHERE ca = (
        SELECT community_area
        FROM CHICAGO_CRIME_DATA
        GROUP BY community_area
        ORDER BY COUNT(*) DESC
        LIMIT 1
    );

 * sqlite:///FinalDB.db
Done.


community_area_name
Austin


## Key Findings & Conclusion

Through the execution of advanced SQL queries across multiple schemas, the following insights were derived regarding the City of Chicago:

### 1. The "Hardship" Link
There is a quantifiable link between the **Hardship Index** and crime volume. Community areas such as **Riverdale (Index: 56.5)** and **Fuller Park** consistently appeared in both the "High Poverty" and "High Crime" query results. This suggests that crime mitigation strategies must be paired with economic support.

### 2. Crime Concentration
Crime is not randomly distributed. The SQL aggregation revealed that specific districts (e.g., District 25) carry a disproportionate load of the city's total crime volume. Resource allocation for policing and community support should be weighted heavily toward these specific geographic identifiers.

### 3. School Safety Dynamics
The analysis of the `CHICAGO_PUBLIC_SCHOOLS` table indicated that "Elementary, Middle, or High Schools" maintain an average safety score of **~49.5**. However, the specific query on school-based crimes highlighted that **Battery** and **Criminal Trespass** are the leading threats to student safety, rather than property crimes.

### Technical Summary
This project demonstrated the capability to:
* Establish connections to SQLite databases within a Python environment.
* Execute complex queries involving **JOINs**, **Sub-queries**, and **Aggregations**.
* Clean and filter data to isolate specific demographics (e.g., Minors, Schools).